src/Gestures/GesturePropertyTransition.vala

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2024 elementary, Inc. (https://elementary.io)
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ * Authored by: Leonhard Kargl <leo.kargl@proton.me>
+ */
+
+/**
+ * A class that will animate a property of a {@link Clutter.Actor} one to one with a gesture or
+ * with easing without a gesture. Respects the enable animation setting.
+ */
+public class Gala.GesturePropertyTransition : Object {
+    public delegate void DoneCallback ();
+
+    /**
+     * The actor whose property will be animated.
+     */
+    public Clutter.Actor actor { get; construct; }
+
+    public GestureTracker gesture_tracker { get; construct; }
+
+    /**
+     * The property that will be animated. To be properly animated it has to be marked as
+     * animatable in the Clutter documentation and should be numeric.
+     */
+    public string property { get; construct; }
+
+    /**
+     * The starting value of the animation or null to use the current value. The value
+     * has to be of the same type as the property.
+     */
+    public Value? from_value { get; construct set; }
+
+    /**
+     * The value to animate to. It has to be of the same type as the property.
+     */
+    public Value to_value { get; construct set; }
+
+    /**
+     * If not null this can be used to have an intermediate step before animating back to the origin.
+     * Therefore using this makes mostly sense if {@link to_value} equals {@link from_value}.
+     * This is mostly used for the nudge animations when trying to switch workspaces where there isn't one anymore.
+     */
+    public Value? intermediate_value { get; construct; }
+
+    private Value actual_from_value;
```

Comment:
I think this requires a comment on how this differs from `from_value`

Reply:
Ah right I wanted to add one but forgot it :sweat_smile: